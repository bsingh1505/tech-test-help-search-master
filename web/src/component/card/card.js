export const CardView = ({ data }) => {
  if (!data) return null;
  return (
    <div className="card">
      <header class="card__header">
        <div>
          <h3>{data?.title}</h3>
          <p>
            {data?.description}{" "}
            <a href={data?.url} target="_blank" rel="noreferrer">
              Click here
            </a>
          </p>
          <p className="slugs">{data?.slug}</p>
        </div>
      </header>
    </div>
  );
};
