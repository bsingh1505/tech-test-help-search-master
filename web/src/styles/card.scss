.card {
  padding: 0.5rem;
  border-radius: 10px;
  margin-bottom: 10px;
  min-height: 70px;
  border: solid 1px;
}

.card__header p {
  opacity: 80%;
  margin: 0;
  padding: 0;
}

.card__header {
  display: flex;
  align-items: flex-start;
  margin-bottom: 10px;
}

.card__header h3 {
  font-size: 15px;
}

.slugs {
  display: inline-flex;
  background-color: orange;
  color: red;
  margin-top: 10px !important;
  padding: 0.4rem !important;
}

.pagination {
  display: flex;
  flex-direction: row;
  align-items: center;
  margin: 1rem;

  Button {
    margin: 0.2rem;
  }
}
