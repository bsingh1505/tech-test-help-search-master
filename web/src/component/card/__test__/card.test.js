import TestRenderer from "react-test-renderer";
import { CardView } from "../card";

const dataWithAllKey = { description: "Description", title: "Title", url: "https://test.com/test", slugs: "test" };
const dataWithMissingKey = { description: "Description", slugs: "test" };

describe("Card component", () => {
  test("match snapshot : With all key", () => {
    const tree = TestRenderer.create(<CardView data={dataWithAllKey} />);
    expect(tree).toMatchSnapshot();
  });
  test("match snapshot : With missing key", () => {
    const tree = TestRenderer.create(<CardView data={dataWithMissingKey} />);
    expect(tree).toMatchSnapshot();
  });
  test("match snapshot : with missing data", () => {
    const tree = TestRenderer.create(<CardView />);
    expect(tree).toMatchSnapshot();
  });
});
