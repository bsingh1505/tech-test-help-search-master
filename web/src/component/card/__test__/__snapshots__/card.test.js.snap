// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component match snapshot : With all key 1`] = `
<div
  className="card"
>
  <header
    class="card__header"
  >
    <div>
      <h3>
        Title
      </h3>
      <p>
        Description
         
        <a
          href="https://test.com/test"
          rel="noreferrer"
          target="_blank"
        >
          Click here
        </a>
      </p>
      <p
        className="slugs"
      />
    </div>
  </header>
</div>
`;

exports[`Card component match snapshot : With missing key 1`] = `
<div
  className="card"
>
  <header
    class="card__header"
  >
    <div>
      <h3 />
      <p>
        Description
         
        <a
          rel="noreferrer"
          target="_blank"
        >
          Click here
        </a>
      </p>
      <p
        className="slugs"
      />
    </div>
  </header>
</div>
`;

exports[`Card component match snapshot : with missing data 1`] = `null`;
